test/systemtest/earlgrey/test_sim_verilator.py

```diff
@@ -196,6 +211,24 @@ def app_selfchecking(request, bin_dir):
 # Therefore, if we do not read quickly enough, we miss the PASS/FAIL indication
 # and the test never finishes.
 
+def assert_selfchecking_test_passes(sim):
+    assert sim.find_in_output(
+        re.compile(r"SW test status changed: SwTestStatusInTest$"),
+        timeout=120,
+        filter_func=utils.filter_remove_sw_test_status_log_prefix
+    ) is not None, "Start of test indication not found."
+
+    log.debug("Waiting for pass string from device test")
+
+    result_match = sim.find_in_output(
```

Comment:
Recommend using the same pass/ fail patterns as DVSim uses for running Verilator tests:
https://github.com/lowRISC/opentitan/blob/f18ff1ef701573f18c4019f8528f13e9187438e2/hw/dv/tools/dvsim/verilator.hjson#L89

Reply:
In this test we want to check for two things:

- Find a line indicating that the test completed.
- Get a success/fail status.

The previous and current approach do so by looking for a line that indicates both the completion and the completion status. The line dvsim is using ("TEST PASSED CHECKS") is only printed on a successful completion, but not otherwise. Hence we'd need another match to check if the test completed. Therefore I think the current solution works better for that use case.

Reply:
The way it is currently hooked up is, if the test passes, it prints `TEST PASSED CHECKS`. If the test fails, it prints `TEST FAILED CHECKS` (which is quite similar to what you are doing with `SW TEST (PASSED|FAILED)`. This is done here:

https://github.com/lowRISC/opentitan/blob/a1b72140bf1a412b264bbf5af22e748f19b82ed4/hw/top_earlgrey/rtl/top_earlgrey_verilator.sv#L276

Apart from SW, there could be checks placed in the SV testbench as well. You currently do not, but you could, in future. The idea is to not only factor in the SW 'vote' but also votes from the testbench. That is why I suggested using this as the pass / fail signature rather that only looking at the SW vote. 

The criteria for a test passing (for dvsim) is at least one passing pattern to be present and none of the failing patterns to be present. This makes the final pass / fail indication more robust since the margin for a passing test is very narrow.